src/main/java/com/amihaiemil/eoyaml/AllYamlLines.java

```diff
@@ -126,23 +140,29 @@ private YamlNode mappingSequenceOrPlainScalar(
             line -> line.trimmed().startsWith("%"),
             line -> line.trimmed().startsWith("!!")
         ).iterator().next();
-        if (first.trimmed().startsWith("-")){
-            node = new ReadYamlSequence(prev, this, guessIndentation);
-        } else if(first.trimmed().contains(":")) {
-            node = new ReadYamlMapping(prev.number(), prev, this,
-                    guessIndentation);
-        } else if(this.original().size() == 1) {
```

Comment:
@newmana  The ticket was concerning the escaping of a scalar in a sequence: why was changing the reading logic necessary?

Reply:
This is probably the central point of the bug - while it is about literals in sequences - part of the problem was that a string like: "arn:foo:bar" would be matched as a mapping.  For both mappings and sequences you need at least one space after the ":" and "-" respectively.

The relevant tests are "returnsValueFromScalarLineWithColon", "returnsValueFromScalarLine", "turnsIntoScalarNotMapping" and turnsIntoScalarNotSequence " - part of this PR.